lib/ood_core/job/adapters/slurm.rb

```diff
@@ -394,10 +394,21 @@ def submit(script, after: [], afterok: [], afternotok: [], afterany: [])
 
           # Set environment variables
           env = script.job_environment || {}
-          unless (script.job_environment.nil? || script.job_environment.empty?)
-            prefix = script.copy_environment? ? "ALL," : "NONE,"  # NONE if false or nil
-            args += ["--export", prefix + script.job_environment.keys.join(",")]
-          end
+          # we default to export NONE, but SLURM defaults to ALL.
+          # we do this bc SLURM setups a new environment loading /etc/profile (modules)
+          # where the PUN (the env sbatch uses) did not.
+          export = if !env.empty? && !script.copy_environment?
```

Comment:
this block should be refactored to a method (private?), something like `def sbatch_export_environment_option(job_environment, copy_environment)` and then just `          args += ["--export", sbatch_export_environment_option(env, script.copy_environment?)]`